### PR TITLE
handle data paths for pages with `getStaticPaths`

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4005** <a href="https://app.graphite.dev/github/pr/vercel/turbo/4005" target="_blank">Graphite</a> 
    * **PR #4013** <a href="https://app.graphite.dev/github/pr/vercel/turbo/4013" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/4013?utm_source=stack-comment).

File: crates/next-core/js/src/internal/page-server-handler.tsx [hunk 1]
```diff
@@ -96,12 +96,19 @@ export default function startHandler({
...
+          /^\/_next\/data\/development(.+).json$/,
+          (_, args) => args
+        );
+      }
       if (
         prerenderFallback === false &&
         // TODO(alexkirsz) Strip basePath.
         !prerenderRoutes.includes(resolvedPath)
       ) {
+        // console.log("prerender not found:", resolvedPath, "   prerendered:", prerenderRoutes);
```
Comment (sokra) [inline crates/next-core/js/src/internal/page-server-handler.tsx:111]:
```suggestion
```

**Review by sokra — APPROVED** [review]

File: crates/next-core/js/src/internal/page-server-handler.tsx [hunk 2]
```diff
@@ -96,12 +96,19 @@ export default function startHandler({
 
       // We provide a dummy base URL to the URL constructor so that it doesn't
       // throw when we pass a relative URL.
-      const resolvedPath = new URL(renderData.url, "next://").pathname;
+      let resolvedPath = new URL(renderData.url, "next://").pathname;
+      if (isDataReq) {
+        resolvedPath = resolvedPath.replace(
+          /^\/_next\/data\/development(.+).json$/,
+          (_, args) => args
```
Comment (jridgewell) [inline crates/next-core/js/src/internal/page-server-handler.tsx:103]:
```suggestion
          "$1"
```

File: crates/next-core/js/src/internal/page-server-handler.tsx [hunk 3]
```diff
@@ -96,12 +96,19 @@ export default function startHandler({
 
       // We provide a dummy base URL to the URL constructor so that it doesn't
       // throw when we pass a relative URL.
-      const resolvedPath = new URL(renderData.url, "next://").pathname;
+      let resolvedPath = new URL(renderData.url, "next://").pathname;
+      if (isDataReq) {
+        resolvedPath = resolvedPath.replace(
```
Comment (alexkirsz) [inline crates/next-core/js/src/internal/page-server-handler.tsx:101]: Can you add a comment to explain why this is necessary/when this can happen?

**Review by alexkirsz — APPROVED** [review]

File: crates/next-core/js/src/internal/page-server-handler.tsx (hunk 1 above)
Comment (ForsakenHarmony) [reply crates/next-core/js/src/internal/page-server-handler.tsx:111]: Really need to get better at reviewing my own PRs 🙃

File: crates/next-core/js/src/internal/page-server-handler.tsx (hunk 2 above)
Comment (jridgewell) [reply crates/next-core/js/src/internal/page-server-handler.tsx]: Regex replace is the most magical API in JS.

**Pull request merged**